### PR TITLE
fix docker (by moving to core version that still works)

### PR DESCRIPTION
**Pull request by toolslive**
- remove use of Core.Time_internal & move to 114.35+03
- update versions of other libs too

### CODE REVIEW

File: src/tools/arakoon_logger.ml [hunk 1]
```diff
@@ -71,30 +71,22 @@ let format_message
       ~ts ~seqnum
       ~section ~level ~lines
   =
+
+  let d =
+    let (f : float) = Obj.magic ts in
```
Comment (domsj) [inline src/tools/arakoon_logger.ml]: do we need to use Obj.magic here?
I would expect there to be another way ...
(and if there isn't: can we specify the type of `ts` so callers can't pass in something of the wrong type?)

File: src/tools/arakoon_logger.ml [hunk 2]
```diff
@@ -71,30 +71,22 @@ let format_message
...
+    let df = mod_float f 1.0 in
+    let udf = df *. 1_000_000. in
+    int_of_float udf
+  in
+  let tz_diff = "+0000" in
   Printf.sprintf
-    "%s - %s - %i/0 - %s - %i - %s - %s"
+    "%s %6d %s %s - %i/0 - %s - %i - %s - %s"
```
Comment (domsj) [inline src/tools/arakoon_logger.ml]: changing the log format?

File: src/tools/arakoon_logger.ml [hunk 3]
```diff
@@ -71,30 +71,22 @@ let format_message
...
-       tm.tm_sec
-       (int_of_float ((mod_float (Core.Time.to_float ts) 1.) *. 1_000_000.))
-       (let open Core_kernel in
-        match offset.Core.Span.Parts.sign with
-        | Sign.Neg -> '-'
-        | Sign.Pos | Sign.Zero -> '+')
-       offset.Core.Span.Parts.hr
-       offset.Core.Span.Parts.min
+     let open Core in
+     Time.format ts "%Y/%m/%d %T" Time.Zone.utc
```
Comment (domsj) [inline src/tools/arakoon_logger.ml]: changing the log format here? (`/` iso `-`, and utc time iso local time + the utc offset)

**Comment by toolslive** [comment]
Hold on, there's something wrong in the microsecond logging part.

File: src/tools/arakoon_logger.ml (hunk 2 above)
Comment (domsj) [reply src/tools/arakoon_logger.ml]: can you make the `/0` part of `%i/0` configurable (while we're breaking backwards compatibility anyway...)?

File: src/tools/arakoon_logger.ml [hunk 4]
```diff
@@ -67,17 +67,23 @@ let console_logger =
 
 
 let format_message
-      ~hostname ~pid ~component
-      ~ts ~seqnum
+      ~hostname ~pid ~thread_id ~component
+      ~(ts:Core.Time.t) ~seqnum
       ~section ~level ~lines
   =
+  let thread_id = 0 in
```
Comment (domsj) [inline src/tools/arakoon_logger.ml]: this binding shadows the argument that's passed in

File: src/tools/arakoon_logger.ml [hunk 5]
```diff
@@ -67,17 +67,23 @@ let console_logger =
...
+     let tm = Unix.localtime fts in
      let offset =
-       Core.Time.utc_offset ~zone:Core.Time.Zone.local ts
-       |> Core.Span.to_parts
+       let zone = Lazy.force Core.Time.Zone.local in
+       Core.Time.utc_offset ~zone ts
+       |> Core.Time.Span.to_parts
      in
      Printf.sprintf
        "%04d-%02d-%02d %02d:%02d:%02d %d %c%02d%02d"
```
Comment (domsj) [inline src/tools/arakoon_logger.ml]: I know it's unrelated, but can you also change the `%d` for printing the microseconds to `%06d`?
Just remembered something about @jeroenmaelbrancke's log parsing having some problems with this, so he would like to see that changed...

File: src/tools/arakoon_logger.ml (hunk 4 above)
Comment (toolslive) [reply src/tools/arakoon_logger.ml]: right.

File: src/tools/arakoon_logger.ml (hunk 5 above)
Comment (toolslive) [reply src/tools/arakoon_logger.ml]: ack

**Review by domsj — APPROVED** [review]

**Pull request merged**